pkg/operation/botanist/dnsrecord.go

```diff
@@ -27,7 +27,7 @@ import (
 func (b *Botanist) DefaultExternalDNSRecord() extensionsdnsrecord.Interface {
 	values := &extensionsdnsrecord.Values{
 		Name:       b.Shoot.GetInfo().Name + "-" + DNSExternalName,
-		SecretName: b.Shoot.GetInfo().Name + "-" + DNSExternalName,
+		SecretName: "dnsrecord-" + b.Shoot.GetInfo().Name + "-" + DNSExternalName,
```

Comment:
How about extracting `"dnsrecord-"` to a constant?

To delete the orphaned secret, you could modify https://github.com/gardener/gardener/blob/247aeda692d432554052392d10a46b5c2ca5342e/pkg/operation/botanist/component/extensions/dnsrecord/dnsrecord.go#L135 to delete the old secret (named like the new one, minus the prefix), ignoring not found errors. You could also add a comment to remove this code in a later release.

Reply:
> How about extracting "dnsrecord-" to a constant?
This makes sense, will do.

> To delete the orphaned secret, you could modify https://github.com/gardener/gardener/blob/247aeda692d432554052392d10a46b5c2ca5342e/pkg/operation/botanist/component/extensions/dnsrecord/dnsrecord.go#L135
This was what @ialidzhikov suggested as well. I'm not sure if I misunderstood how secrets are used by the different DNSRecords, but it seems that secrets are used by more than one DNSRecord. For example, the `external` secret is used by the `ingress` DNSRecord as well as the `external` DNSRecord. The `internal` secret is used by the `owner` DNSRecord as well as the `internal` DNSRecord. Can we really remove a secret after deploying only one of these records while the other might still be referencing the old secret? Especially seeing how far apart e.g. deploying the `ingress` record is from the `external` one during the Shoot reconciliation?

Reply:
This is a good question. I think it should be ok, since reconciliations of `DNSRecord` resources, similarly to most other extension resources, only happen if they are annotated the `operation=reconcile`, but at the moment this annotation is added, the `secretRef` will be corrected as well. A reconciliation of `ingress` taking place at the same time or shortly after `external` is reconciled should only be possible if it has failed for whatever reason and is in a retry with backoff loop, but I also think in this case a failed reconciliation because of an invalid `secretRef` wouldn't cause any extra trouble, the resource will have to be updated anyway to make it succeed.

I would of course recommend not to take my assumptions for granted and test this, including some manually forced reconciliations of resources with invalid `secretRef` just in case.

Reply:
Speaking of the `ingress` DNSRecord @voelzmo, don't we need an adaption [here](https://github.com/gardener/gardener/blob/97e2e316891bfad3c7e381d7fea7d068f7453aa9/pkg/operation/botanist/addons.go#L141) as well?

Reply:
> Speaking of the ingress DNSRecord @voelzmo, don't we need an adaption here as well?

Yeah, sorry, all of this is already on my branch – I thought I had pushed this already. I'll add it for the sake of discussion, although Stoyan already suggested a different approach, just for the sake of discussion.